src/graphnet/models/graphs/utils.py

```diff
@@ -158,3 +163,36 @@ def cluster_summarize_with_percentiles(
         )
 
     return array
+
+
+def ice_transparency(datum: int = 1950) -> Tuple[interp1d, interp1d]:
+    """Calculate the normalized scattering and absorption lengths.
+
+    Values are calculated for iceCube ice as a function of depth.
+
+    Args:
+        datum: The datum depth in meters.
+               Default to 1950.
+
+    Returns:
+        f_scattering: Function that takes a normalized depth
+                      and returns the corresponding normalized
+                      scattering length.
+        f_absorption: Function that takes a normalized depth
+                      and returns the corresponding normalized
+                      absorption length.
+    """
+    # Data from page 31 of https://arxiv.org/pdf/1301.5361.pdf
+    # Datum is from footnote 8 of page 29
+    df = pd.read_parquet(
+        os.path.join(DATA_DIR, "ice_properties/ice_transparency.parquet"),
+    )
+    df["z"] = df["depth"] - datum
```

Comment:
I think the name "datum" is incredible confusing and I highly suspect it's the result of a machine translation going wrong. I realize you did not name this variable, but let's take this opportunity to get it right.  

From the code, I think it's clear that this variable is an offset in the scaling `z = (depth - offset)/500`. I believe this is the 2nd place teams attempt to transform global coordinates into icecube coordinates, such that the resulting interpolation functions can be evaluated using the scaled z-coordinate of the pulses. I.e `z = depth - offset` gives you the z-coordinate in icecube coordinates and the `/500` gives you the standardized feature, identical to the standardized z-coordinate of the pulses ( see the transform [here](https://github.com/graphnet-team/graphnet/blob/ee2addc6f495c68262f3fe36bea4a36b4ccabd98/src/graphnet/models/detector/icecube.py#L35)) . This means, however, as this code is written, that this function expects a very specific standardization in order to be correct (`/500`), which I think is problematic, because if one were to pass the `IceCubeDeepCore` detector instead, the scaling is (`/100`) see [here](https://github.com/graphnet-team/graphnet/blob/ee2addc6f495c68262f3fe36bea4a36b4ccabd98/src/graphnet/models/detector/icecube.py#L93).

Reply:
I think for the purposes of this PR, you could just slightly change the function - to make it clear that this is a potential problem. Like so:

```python
def ice_transparency() -> Tuple[interp1d, interp1d]:
    """ Returns interpolation functions for optical properties of IceCube.

        NOTE: The resulting interpolation functions assumes that the 
        Z-coordinate of pulse are scaled as `z = z/500`. 
        Any deviation from this scaling method results in inaccurate results. 

    Returns:
        f_scattering: Function that takes a normalized depth
                      and returns the corresponding normalized
                      scattering length.
        f_absorption: Function that takes a normalized depth
                      and returns the corresponding normalized
                      absorption length.
    """
    # Data from page 31 of https://arxiv.org/pdf/1301.5361.pdf
    # Datum is from footnote 8 of page 29
    df = pd.read_parquet(
        os.path.join(DATA_DIR, "ice_properties/ice_transparency.parquet"),
    )
    df["z"] = df["depth"] - 1950 # Should actually be 1948.2
    df["z_norm"] = df["z"] / 500
    df[
        ["scattering_len_norm", "absorption_len_norm"]
    ] = RobustScaler().fit_transform(df[["scattering_len", "absorption_len"]])

    f_scattering = interp1d(df["z_norm"], df["scattering_len_norm"])
    f_absorption = interp1d(df["z_norm"], df["absorption_len_norm"])
    return f_scattering, f_absorption
```

I think that would be OK as we currently do not have an easy way of including this information. In a future iteration, we can think of a more structured way of including this information, and to deal with the hard-coded feature names.

Reply:
You are right, this function was barely modified compared to the one used in [icecube-2nd-place](https://github.com/DrHB/icecube-2nd-place).

I share your concern about the lack of generalization of the ice properties in general. 
Maybe it is a good idea to make the small changes you suggested, and have it deleted before the _GraphNet 2.0.0_ release